### DISCUSSION TITLE
AF_XDP support

### DISCUSSION THREAD

**Discussion by tbarbette** [tbarbette/fastclick#302] [Ideas]
While DPDK is adding support for AF_XDP (http://mails.dpdk.org/archives/dev/2018-February/091502.html) we should have a FromXDPDevice and ToXDPDevice element (or make it a sub-method of FromDevice if the push/pull syntax is identical)  that allows doing high-speed without using DPDK.
The setup of DPDK stays a little bit complex, requires to bind some NICs, etc. This would allow somehow much higher-speed processing in user-level, without the complexity of DPDK.

**Comment by ahenning** [comment]
If someone is interested in creating this element and have time available please contact me.
